README.md

```diff
@@ -1,30 +1,29 @@
-# Glutil
+# glutil
 
-A collection of utilities for managing AWS Glue partitions.
+A collection of utilities for managing partitions of tables in the AWS Glue Data Catalog that are built on datasets stored in S3.
 
 ## Hello Friends
 
-Glutil is currently in the process of being open sourced.
+`glutil` is currently in the process of being open sourced.
 There are still a couple small things we want to get done before calling that job done.
 As such, please pardon any sharp edges, and let us know about them by [creating an issue](https://github.com/Journera/glutil/issues/new).
 
 ## Background
 
-Amazon's Glue Catalog is generally pretty great, but sometimes things don't play well together, or a configuration mistake was made.
-For those cases, we have these utilities.
+AWS's Glue Data Catalog provides an index of the location and schema of your data across AWS data stores and is used to reference sources and targets for ETL jobs in AWS Glue. It is fully-integrated with AWS Athena, an ad-hoc query tool that uses the Hive metastore to build external tables on top of S3 data and PrestoDB to query the data with standard SQL. Journera heavily uses Kinesis Firehoses to write data from our platform to S3 in near real-time, Athena for ad-hoc analysis of data on S3, and Glue's serverless engine to execute PySpark ETL jobs on S3 data using the tables defined in the Data Catalog. Using the Data Catalog is generally pretty great, but sometimes all of these managed services don't play well together, or a configuration mistake was made (e.g., in a table DDL). For those cases, we have these utilities.
 
-At Journera, our original use case for this project was as a Glue Crawler replacement for tables that don't use Hive-compliant path structures.
-For the most part this is a workaround, because at present Terraform (which we use to manage our Firehoses) does not support using formatted prefixes with data written in JSON.
+At Journera, our original use case for this project was as a Glue Crawler replacement for adding new partitions to tables that don't use Hive-style partitions and for tables built on top of S3 datasets that the Glue Crawler could not successfully parse.
+For the most part this is a workaround, because of current limitations with the Glue Crawler and Terraform, which does not support configuring Kinesis Firehoses to write JSON data to S3 using formatted prefixes.
```

Comment:
I don't think we're encountering glue crawler limitations, because we're unable to fully use the glue crawler because of the terraform limitations.

Unless we've confirmed using the catalog table crawler source or using a json-formatted data set with hive-style partitions doesn't work with the catalog, I don't think we can say these are glue crawler limitations.

Reply:
We specifically encounter Glue crawler limitations for the customer event archive, where we have CustomerAddress events with address token counts. Because we cannot exclude Glue crawling a specific JSON field, the crawler attempts to index every key in this map and breaks.